src/Reflection/ReflectionHelper.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Leadin\SurvivalKitBundle\Reflection;
+
+class ReflectionHelper
+{
+    /**
+     * Returns class name without namespace
+     *
+     * @param object|string $object
+     * @return string
+     */
+    public static function getClassShortName($object): string
+    {
+        return (new \ReflectionClass($object))->getShortName();
+    }
+
+    public static function getNamespace($object): string
```

Comment:
Not sure we should add @throws for all possible exceptions, I think would be better to add them only when exception is explicitly thrown. Otherwise our docs will be out of date soon. I think it's something to discuss.

Reply:
> to add them only when exception is explicitly thrown

I do not want to deep into details (to be a boring person), but exceptions could be thrown deeply in call stack. And Ide showed them to me unhandled (I use storm). I think there is a plugin for VS too. We tried to discuss it in our last meeting about entities. @leadinpier @a-bastien

Reply:
According to https://wiki.youdge.com/doku.php?id=coding_standars#comments method docblock is mandatory for public function or other functions throwing exceptions.

Reply:
That's the point, this method is not throwing any exception. I'm close to this opinion https://www.php.net/manual/en/class.reflectionexception.php#123784

Reply:
Maybe the end of the day - and I do not understand the way. [Here is code](https://onlinephp.io?s=s7EvyChQ4OXi5SopqlSo5uXiVClRsFXISy1XCEpNy0lNLsnMz3POSSwu1lDPzCtLzMlMUUgsSi_NTc0rUde05uWqVUhOLEnOUNBAKHetSE4tADEUVFI1kcwszckBaQBZVpZYFJ9SmlugoVKiaQ0A&v=7.1.33) , [here is manual, see throws section](https://www.php.net/manual/en/reflectionclass.construct). What is this comment about? Is this about, that such types of exception must be caught at the top of application? Why the throw tag in phpdoc is wrong?

Reply:
As far as I understand it, ReflectionClass::__construct() can throw a ReflectionException if the class to reflect does not exists as explained here at https://www.php.net/manual/en/reflectionclass.construct.php.  And therefore, we shall either surround the instruction building a new ReflectionClass or add the ReflectionException within a @throw tag in the docblock according to our coding standards.

Reply:
> What is this comment about? Is this about, that such types of exception must be caught at the top of application? Why the throw tag in phpdoc is wrong?

You can divide exceptions into two types - checked and unchecked (inspiration from Java).
Checked exceptions are expected to be caught/handled by the code at the call site of a function that throws a checked exception, for these exceptions we should add `@throws` in the docblock, saying to catch/handle it.

On the other hand, Unchecked exceptions are not welcome when they’re thrown, and they’re not supposed to be caught and handled because they present an unrecoverable state of the application. We don’t know how we should continue the execution if an unchecked exception is thrown, examples like DB connection failed, PHP errors. For these exceptions we should not add `@throws` in the docblock, saying to do not catch it. It’s better to let them bubble up to the topmost layer of the code, log them, alert the developers, and fix the cause asap.

Summarizing `ReflectionException` is unchecked exception so we should not add @throws for it.

Reply:
Thank You for the explanation! ([alternative explanation](https://phpstan.org/blog/bring-your-exceptions-under-control)) Maybe it is time to make it legal in wiki and [ide](https://stackoverflow.com/questions/74325916/how-to-disable-warning-for-specific-exceptions)